### ISSUE TITLE
NaN-Check in TraCI

### ISSUE THREAD

**Issue by der-b** [eclipse-sumo/sumo#4909]
tl;dr
Using TraCI to crate polygons with shape-positions containing NaNs will terminate SUMO with a error message similar to "pure virtual function called". Parameter checks for NaNs in TraCI would prevent this.

The long story:
hi,

we heavily use polygons to visualize data within sumo. During the
development of the visualization we run into an error which terminated
sumo with an error message similar to: "pure virtual function called".

The reason was a error in one of our calculations for the shapes used
as polygons. The position of the shapes was filled with NaNs which we
provided to sumo via TraCI. I didn't locate the code where the error
ultimately arises, but i guess it has something to do with the used
RTree which sorts the polygons by their positions. Since relational
operators for NaNs always fails, i guess that this cause the error
somehow.

Due to the error message i suspected an memory handling error within
sumo. I wasted a lot of time before i noticed the NaNs. Therefore as
an enhancement i would suggest to add NaN checks in the implementation
of TraCI. A proper error message would prevent other developers to run
into the same issues.

Unfortunately I'm currently not able to provide a sample code.

**Comment by namdre** [comment]
I literally bombarded sumo-gui with random polygons containing variable amounts of NaNs (with the python client on Linux) but couldn't make the release build it crash. The debug build always hit an assertion in the RTree and then dumped core. 
What platform / buildType / traciClient did you use?
